ingestion/src/metadata/ingestion/source/pipeline/flink/metadata.py

```diff
@@ -0,0 +1,166 @@
+
+
+class FlinkSource(PipelineServiceSource):
+    """
+    Implements the necessary methods ot extract
+    Pipeline metadata from Flink's REST API
+    """
+
+    @classmethod
+    def create(
+        cls, config_dict, metadata: OpenMetadata, pipeline_name: Optional[str] = None
+    ):
+        config: WorkflowSource = WorkflowSource.parse_obj(config_dict)
+        connection: FlinkConnection = config.serviceConnection.root.config
+        if not isinstance(connection, FlinkConnection):
+            raise InvalidSourceException(
+                f"Expected FlinkConnection, but got {connection}"
+            )
+        return cls(config, metadata)
+
+    def get_connections_jobs(self, pipeline_details: FlinkPipeline):
+        """Returns the list of tasks linked to connection"""
+        pipeline_info = self.client.get_pipeline_info(pipeline_details)
+        tasks = pipeline_info.get("vertices", [])
+        om_tasks = []
+        for task in tasks:
+            om_tasks.append(
+                Task(
+                    name=f"{task.get('name')}_{task.get('id')}",
+                ),
+            )
+        return om_tasks
+
+    def yield_pipeline(
+        self, pipeline_details: FlinkPipeline
+    ) -> Iterable[Either[CreatePipelineRequest]]:
+        """
+        Convert a Connection into a Pipeline Entity
+        :param pipeline_details: pipeline_details object from Flink
+        :return: Create Pipeline request with tasks
+        """
+        pipeline_request = CreatePipelineRequest(
+            name=pipeline_details.name,
+            service=self.context.get().pipeline_service,
+            sourceUrl=self.get_source_url(pipeline_details),
+            tasks=self.get_connections_jobs(pipeline_details),
+        )
+        yield Either(right=pipeline_request)
+        self.register_record(pipeline_request=pipeline_request)
+
+    def get_pipelines_list(self) -> Iterable[FlinkPipeline]:
+        """Get List of all pipelines"""
+        for pipeline in self.client.get_jobs():
+            yield FlinkPipeline(**pipeline)
+
+    def get_pipeline_name(self, pipeline_details: FlinkPipeline) -> str:
+        return pipeline_details.name
+
+    def yield_pipeline_lineage_details(
+        self, pipeline_details: Any
+    ) -> Iterable[Either[AddLineageRequest]]:
+        """Get lineage between pipeline and data sources"""
+
+    def yield_pipeline_status(
+        self, pipeline_details: FlinkPipeline
+    ) -> Iterable[Either[OMetaPipelineStatus]]:
+        """
+        Get Pipeline Status
+        """
+        try:
+            task_status = []
+            for task in self.client.get_pipeline_info(pipeline_details).get(
+                "vertices", []
+            ):
+                task_status.append(
+                    TaskStatus(
+                        name=task.get("id"),
+                        executionStatus=StatusType.Successful,
+                        startTime=Timestamp(task.get("start-time")),
```

Comment:
Here, not sure but should we have to convert `task.get("start-time")` into Milliseconds ?

Reply:
It's into miliseconds format from API itself.